aten/src/ATen/core/Dict_inl.h

```diff
@@ -19,6 +23,27 @@ inline bool shallowEquals(const IValue& lhs, const IValue& rhs) {
     AT_ERROR("shallowEquals(IValue, IValue) not implemented for type ", lhs.tagKind());
   }
 }
+
+template<class Key, class Value>
+Dict<Key, Value> toTypedDict(GenericDict dict) {
+  if (dict.impl_->elementTypes.has_value()) {
+    // TODO Show types in error message: TORCH_INTERNAL_ASSERT(getTypePtr<Key>() == dict.impl_->elementTypes->keyType, "Tried to cast a Dict with key type ", (*dict.impl_->elementTypes->keyType)->str(), " to a dict with different key type ", getTypePtr<Key>()->str());
```

Comment:
what prevents us from this TODO?

Reply:
An include cycle that means we don't have the str() method on jit types available here.

It can he fixed by splitting jit_type.h into a jit_type.h and jit_type_inl.h but that is a lot of extra effort I didn't wanna do in this PR.

Reply:
I found a way to make it work